File: frontend/bookstore-app/src/types/cuota.ts
export interface PagoCuotaDto {
  id: number;
  nroReferencia: string;
  fecha: string;
  importe: number;
}

export interface CuotaListado {
  id: number;
  comprobanteId: number;
  numeroComprobante?: string;
  fechaComprobante?: string;
  clienteId: number;
  clienteNombre?: string;
  zonaId?: number;
  zonaNombre?: string;
  fechaCuota?: string;
  importe: number;
  importePagado: number;
  estado?: string;
  numeroCuota: number; // 0 = contraentrega, 1+ = cuotas regulares
  esCuotaCero: boolean; // Calculado en el backend: numeroCuota === 0
  pagos: PagoCuotaDto[];
}

export interface CreatePagoCuotaDto {
  nroReferencia: string;
  fecha: string;
  importe: number;
}

export interface CreatePagoComprobanteDto {
  nroReferencia: string;
  importe: number;
  fecha?: string;
}

File: backend/BookstoreAPI/Models/Cuota.cs
namespace BookstoreAPI.Models
{
    public class Cuota
    {
        public int Id { get; set; }
        public int? Comprobante_Id { get; set; }
        public int NumeroCuota { get; set; } // 0 = contraentrega, 1+ = cuotas regulares
        public DateTime? Fecha { get; set; }
        public decimal? Importe { get; set; }
        public decimal? ImportePagado { get; set; }
        public string? Estado { get; set; }
    }
}
